ios/Classes/AnnotationController.swift

```diff
@@ -97,21 +97,26 @@ class FlutterAnnotation: NSObject, MKAnnotation {
     var isDraggable: Bool?
     var wasDragged: Bool = false
     var isVisible: Bool? = true
-    var icon: AnnotationIcon = AnnotationIcon.init()
+    var icon: AnnotationIcon
+    var anchor: CGPoint
     
-    public init(fromDictionary annotationData: Dictionary<String, Any>) {
+    public init(fromDictionary annotationData: Dictionary<String, Any>, registrar: FlutterPluginRegistrar) {
         let position :Array<Double> = annotationData["position"] as! Array<Double>
         let lat: Double = position[0]
         let long: Double = position[1]
+        let id: String = annotationData["annotationId"] as! String
         self.coordinate = CLLocationCoordinate2D(latitude: lat, longitude: long)
         let infoWindow :Dictionary<String, Any> = annotationData["infoWindow"] as! Dictionary<String, Any>
         self.title = infoWindow["title"] as? String
         self.subtitle = infoWindow["snippet"] as? String
         self.id = annotationData["annotationId"] as? String
         self.isVisible = annotationData["visible"] as? Bool
+        let anchorData = annotationData["anchor"] as! Array<Float>
+        self.anchor = CGPoint(x: CGFloat(anchorData[0]), y: CGFloat(anchorData[1]))
         if let alpha :Double = annotationData["alpha"] as? Double {
             self.alpha = alpha
         }
+        self.icon = FlutterAnnotation.getAnnotationIcon(registrar: registrar, iconData: annotationData["icon"] as! Array<Any>, annotationId: id)
```

Comment:
`annotationData["icon"]` can be nil and therefore we should check that, probably in `getAnnotationIcon`, especially as the Flutter function `fromAssetImage()` is async and therefore it is highly likely, that an appleMap is build without the icon data present.